nncf/ptq/openvino/quantization.py

```diff
@@ -0,0 +1,145 @@
+    """
+    ir_model_xml = '/tmp/model.xml'
+    ir_model_bin = '/tmp/model.bin'
+    openvino.runtime.serialize(model, ir_model_xml, ir_model_bin)
+
+    model_config = {
+        'model_name': 'model',
+        'model': ir_model_xml,
+        'weights': ir_model_bin,
+    }
+
+    engine_config = {
+        'device': 'CPU',
+        'stat_requests_number': 1,
+        'eval_requests_number': 1,
+    }
+
+    algorithms = [
+        {
+            'name': 'DefaultQuantization',
+            'params': {
+                'target_device': target_device,
+                'preset': preset,
+                'stat_subset_size': subset_size,
+                'use_fast_bias': fast_error_correction,
+                'model_type': model_type,
+            }
+        }
+    ]
+
+    pot_model = pot.load_model(model_config)
+    pot_dataloader = SizedNNCFDataLoaderImpl(calibration_dataset,
+                                             calibration_dataset._transform_fn,
+                                             calibration_dataset.batch_size)
+    engine = CustomEngine(engine_config, pot_dataloader, pot_dataloader)
+    pipeline = pot.create_pipeline(algorithms, engine)
+    compressed_model = pipeline.run(pot_model)
+    pot.compress_model_weights(compressed_model)
+
+    compressed_model_paths = pot.save_model(compressed_model, save_path='/tmp/pot', model_name='model')
+    ir_model_xml = compressed_model_paths[0]['model']
+    ir_model_bin = compressed_model_paths[0]['weights']
+    ie = openvino.runtime.Core()
+    quantized_model = ie.read_model(model=ir_model_xml, weights=ir_model_bin)
+
+    return quantized_model
+
+
+def quantize_with_accuracy_control_impl(model: openvino.runtime.Model,
+                                        calibration_dataset: NNCFDataLoader,
+                                        validation_dataset: NNCFDataLoader,
+                                        validation_fn: Callable[[openvino.runtime.Model, NNCFDataLoader], float],
+                                        max_drop: float = 0.01,
+                                        higher_better: bool = True,
+                                        preset: str = 'performance',
+                                        target_device: str = 'ANY',
+                                        subset_size: int = 300,
+                                        fast_error_correction: bool = True,
+                                        model_type: Optional[str] = None) -> openvino.runtime.Model:
+    """
+    Implementation of the `quantize_with_accuracy_control()` method for the OpenVINO backend.
+    """
+    ir_model_xml = '/tmp/model.xml'
+    ir_model_bin = '/tmp/model.bin'
+    openvino.runtime.serialize(model, ir_model_xml, ir_model_bin)
+
+    model_config = {
+        'model_name': 'model',
+        'model': ir_model_xml,
+        'weights': ir_model_bin,
+    }
+
+    engine_config = {
+        'device': 'CPU',
+        'stat_requests_number': 1,
```

Comment:
Based on the recent experience, I would recommend setting numbers of requests to `2` at least.

Reply:
Done